packages/schema/src/types/hooks.ts

```diff
@@ -19,6 +20,9 @@ export type HookResult = Promise<void> | void
 export type TSReference = { types: string } | { path: string }
 
 export type WatchEvent = 'add' | 'addDir' | 'change' | 'unlink' | 'unlinkDir'
+export type VueTSConfig = TSConfig & {
+  vueCompilerOptions?: VueCompilerOptions;
+};
```

Comment:
If the user doesn't have this dependency installed, this (and mention above in schema) will be typed as any. We could use something like this type:

```ts
type IsAny<T> = 0 extends 1 & T ? true : false
```

to conditionally augment tsConfig only when `VueCompilerOptions` is not any...

Reply:
@danielroe changes made!
I've been pondering about it for some days now but I still don't quite understand one thing: how can the user not have the dependency installed if they're installing nuxt? 🤔
Also, why don't we need to make the same check for other dependencies as well?

Reply:
The user will only have this dependency installed if they add `vue-tsc`. Nuxt doesn't ship with `vue-tsc` by default.

Reply:
Oh, I see! Thank you 😄